tests/components/nx584/test_binary_sensor.py

```diff
@@ -153,11 +153,28 @@ def test_nx584_zone_sensor_normal():
     assert not sensor.should_poll
     assert sensor.is_on
     assert sensor.extra_state_attributes["zone_number"] == 1
+    assert not sensor.extra_state_attributes["bypassed"]
 
     zone["state"] = False
     assert not sensor.is_on
 
 
+def test_nx584_zone_sensor_bypassed():
+    """Test for the NX584 zone sensor."""
+    zone = {"number": 1, "name": "foo", "state": True, "bypassed": True}
+    sensor = nx584.NX584ZoneSensor(zone, "motion")
```

Comment:
Normally we require integrations to be set up through the core APIs (instead of testing the class directly).

But, I see everything in these tests still does that, so I think it's OK to exempt this one.

If you are looking for ways to help improve this integration, this might be a nice next step in improve the quality of this integration.

Reply:
Ok, I will try to improve this integration tests. Moreover, I have some new features I think will be useful.
I will check other integration tests before.

Thanks you, for letting me contribute!